internal/logic/logic.go

```diff
@@ -1,34 +1,185 @@
+		loginService: loginSvc,
+		jwtService:   jwtService,
+		msgQueue:     msgQueue,
 	}
 }
 
 func (l userMgmtSvcLogic) HealthCheck() bool {
 	// check all internal services are working fine
 	return l.DsSvc.HealthCheck()
 }
+
+func (l userMgmtSvcLogic) Signup(credential model.SignUpCredentials) *respModel.Response {
+	if credential.Email == "" || credential.Password == "" || credential.Name == "" {
+		return &respModel.Response{
+			Status:  http.StatusBadRequest,
+			Message: "Provide required details",
+			Data:    nil,
+		}
+	}
+	result, err := l.DsSvc.Get(map[string]interface{}{"email": credential.Email})
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: "Problem creating account",
+			Data:    nil,
+		}
+	}
+	if len(result) != 0 {
+		log.Error("Email is already in use")
+		return &respModel.Response{
+			Status:  http.StatusBadRequest,
+			Message: "Email is already in use",
+			Data:    nil,
+		}
+	}
+
+	newUser := model.User{
+		Id:           uuid.New().String(),
+		Email:        credential.Email,
+		Password:     credential.Password,
+		Name:         credential.Name,
+		Company:      "perennial",
+		RegisteredOn: credential.RegistrationTimestamp,
+	}
+	salt := bCrypt.GeneratePasswordHash([]byte(newUser.Password))
+	newUser.Salt = salt
+	err = l.DsSvc.Insert(newUser)
+	if err != nil {
+		log.Error(err.Error())
+		return &respModel.Response{
+			Status:  http.StatusBadRequest,
+			Message: "Problem creating an account",
+			Data:    nil,
+		}
+	}
+	userID := fmt.Sprintf(`{"user_id":"%s"}`, newUser.Id)
+	err = l.msgQueue.MsgBroker.PushMsg(userID, l.msgQueue.PubId, l.msgQueue.Channel)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: "Problem notifying account management service",
+			Data:    nil,
+		}
+	}
+
+	return &respModel.Response{
+		Status:  http.StatusCreated,
+		Message: "SUCCESS",
+		Data:    nil,
+	}
+}
+
+func (l userMgmtSvcLogic) Login(w http.ResponseWriter, credential model.LoginCredentials) *respModel.Response {
+	if credential.Email == "" || credential.Password == "" {
+		return &respModel.Response{
+			Status:  http.StatusBadRequest,
+			Message: "Provide required details",
+			Data:    nil,
+		}
+	}
+	result, err := l.DsSvc.Get(map[string]interface{}{"email": credential.Email})
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: "Problem logging into your account",
+			Data:    nil,
+		}
+	}
+	if result[0].Email == "" {
+		return &respModel.Response{
+			Status:  http.StatusUnauthorized,
+			Message: "User account was not found",
+			Data:    nil,
+		}
+	}
+	if result[0].Active != true {
+		return &respModel.Response{
+			Status:  http.StatusAccepted,
+			Message: "Problem logging into your account",
+			Data:    "Account activation in progress",
+		}
+	}
+
+	hashedPassword := []byte(result[0].Salt)
```

Comment:
Combine the salt for the email(obtained from DB) with the password in the request.
Pass this new password through the encryption flow to get the encrypted password.
Get the record for this user with this email and new generated password from the DB and return appropriate error if no records are found.

Reply:
done